### ISSUE TITLE
Deprecation notice

### ISSUE THREAD

**Issue by paales** [Smile-SA/elasticsuite#1680]
### Preconditions
Magento Version: M2.3.3 CE
ElasticSuite Version: Elastic 6.8.x

### Steps to reproduce
Load a category page

### Expected result
No deprecation notices

### Actual result

```
elasticsearch_1  | [2020-01-22T21:10:49,954][WARN ][o.e.d.s.s.FieldSortBuilder] [Zsvh-JE] [nested_path] has been deprecated in favor of the [nested] parameter
elasticsearch_1  | [2020-01-22T21:10:49,954][WARN ][o.e.d.s.s.FieldSortBuilder] [Zsvh-JE] [nested_filter] has been deprecated in favour for the [nested] parameter
```

Minor issue, but fix should be relatively easy.

https://github.com/Smile-SA/elasticsuite/blob/d3c00e273c6b7a8ed2c495a7d4fcb9ba89a82d68/src/module-elasticsuite-core/Search/Adapter/Elasticsuite/Request/SortOrder/Builder.php#L66-L87

**Comment by romainruaud** [comment]
Hi @paales ,

thank you for reporting it.

By the way, there might be other locations where we do use nested_path, @rbayet can you have a look at it ? (and also ensure that `nested` is compatible with ES 5.x, otherwise, we'd have to wait for our drop of ES 5.x support to handle this).

Regards

**Comment by rbayet** [comment]
Hello @paales, @romainruaud,

This is the only location in the code where we use `nested_path` and `nested_filter` (and the corresponding unit testing code, of course).

Indeed, instead of using
```
  "sort": [
    {
      "category.position": {
        "order": "asc",
        "missing": "_last",
        "unmapped_type": "keyword",
        "nested_path": "category",
        "mode": "min",
        "nested_filter": {
          "terms": {
            "category.category_id": [
              "4"
            ],
            "boost": 1
          }
        }
      }
```

the following syntax works flawlessly on a ES 6.x : 
```
  "sort": [
    {
      "category.position": {
        "order": "asc",
        "missing": "_last",
        "unmapped_type": "keyword",
        "nested": {
            "path": "category",
            "filter": {
              "terms": {
                "category.category_id": [
                  "4"
                ],
                "boost": 1
              }
            }
        },
        "mode": "min"
      }
    },
```

**Sadly**, as expected from the [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-request-sort.html#nested-sorting), **it is not supported by ES 5.x**.
On a 5.6.14 : 
```
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "[field_sort] unknown field [nested], parser not found"
      }
    ]
```

Regards,

**Comment by romainruaud** [comment]
Well then we could go with a Deprecation plugin for ES 5, something like this : https://github.com/Smile-SA/elasticsuite/blob/b4706d5e6e1da5e311e15738d853a4ca20ba20e3/src/module-elasticsuite-core/Plugin/Deprecation/Client/ClientPlugin.php

But it could be complicated to write it.

Imho, the best way is to address this issue for the 2.9 release when we will add ES 7 support and remove the ES 5 support.

Regards

**Comment by romainruaud** [comment]
Fixed during the implemenration of #1773 

Will be part of next 2.9.0 release, thank you for reporting it
